ERCS/erc-7660.md

```diff
@@ -0,0 +1,426 @@
+
+Implementers of this standard **MUST** have all of the following functions:
+
+```solidity
+pragma solidity ^0.8.0;
+
+interface IERC7660 {
+
+    // Struct
+
+     struct VestInfo {
+        uint256 total;
+        uint256 released;
+        uint256 startTime;
+      }
+    
+    // Events
+
+    /**
+     * @dev Event emitted when a message is sent.
+     * @param sender The address of the sender will exclude vesting or not.
+     * @param flag  The flag whether the sender address will exclude vesting or not;
+     */
+     event SetExcludedVest(address sender,bool flag);
+     
+    /**
+     * @dev Event emitted when a message is sent.
+     * @param sender The address of the uniswap Router will exclude vesting or not.
+     * @param flag  The flag whether the uniswap Router will exclude vesting or not;
+     */ 
+      event SetSwapRouter(address sender,address swapRouter, bool flag);
+
+    // Functions
+
+    /**
+     * @dev Function to get a user's Linear Release token info.
+     * @param account The address of the user.
+     * @return total The total token will be released in the period.
+     * @return canRelease The current will be released token in the period.
+     * @return released The token has be released in the period.
+     */
+    function getCanReleaseInfo(address account) public view returns (uint256 total, uint256 canRelease, uint256 released);
+
+    /**
+     * @dev Function to claim the user's released token to the account.
+     * @param account The address of the user.
+     */
+    function claimRelease(address account) external;
+     
+    /**
+     * @dev Function to set the accout for exclude vesting.
+     * @param account The address of the sender will exclude vesting or not.
+     * @param flag  The flag whether the sender address will exclude vesting or not.
+     */
+    function setExcludedVest(address account,bool flag) external; 
+    
+    /**
+     * @dev Function to set the uniswap Router address which will receive token without vesting.
+     * @param swapRouter The address of the uniswap router.
+     * @param flag  The flag whether the router address will exclude vesting or not.
+     */
+    function setSwapRouter(address swapRouter, bool flag) external;
+
+    /**
+     * @dev Function to retrieve a user's vest token info in the each circle.
+     * @param account The address of the user.
+     * @param circleIndex The user vest token circle index.
+     * @return total The total token will be released in the period for the user.
+     * @return released  The token has be released in the period for the user.
+     * @return startTime The token will be released at the start time .
+     */
+    function userVestInfo(address account,uint256 circleIndex) external view returns (uint256 total, uint256 released,uint256 startTime);
+}
+```
+
+## Rationale
+
+There are four major improvements in this protocol:
```

Comment:
The Rationale section is here to explain _why_ you made particular choices while designing your proposal. My favourite analogy is:

> Motivation: We need to build a shed because...
> Rationale: The shed needs to be red because...